core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushLimitThroughSemiJoin.java

```diff
@@ -33,6 +34,7 @@
 
     // Applies to both limit with ties and limit without ties.
     private static final Pattern<LimitNode> PATTERN = limit()
+            .with(isOrderSensitive().equalTo(false))
```

Comment:
Why? As long as the ordering doesn't depend on the column emitted by SemiJoin, it should be legal to push down. SemiJoin doesn't change the ordering of its input

Reply:
Making this change.
Should there be a PushTopNThroughSemiJoin rule as well which pushes down TopN if it's orderingScheme doesn't depend on the column emitted by SemiJoin ?